modules/economy.js

```diff
@@ -1,295 +1,255 @@
-          }
-        })
-        client.pool.query('UPDATE `guildEconomyUserBank` SET inventory = ? WHERE guild = ? AND member = ?', [JSON.stringify(inventoryData), guild.id, member.id])
+      } else {
+        return false
       }
     })
   },
-  itemActivate: (client, member, guild, item, callback) => {
-    item.productMeta = JSON.parse(item.productMeta)
+  executeItemFunctions: (client, member, guild, shopProduct, callback) => {
+    // TODO: Change the column name from "productMeta" to "productProperties". The code changes will be commented and highlighted.
 
-    const action = item.productMeta.action
-    const userInputRequirements = item.productMeta.properties
-    const userInputs = {}
-    if ((Object.prototype.hasOwnProperty.call(item.productMeta, 'singlebuy') && item.productMeta.singlebuy)) {
-      let propertiesString = ''
+    const productProperties = JSON.parse(shopProduct.productMeta)
+    // * const productProperties = JSON.parse(shopProduct.productProperties)
 
-      if (userInputRequirements && userInputRequirements.length > 0) {
-        if (item.userInput) {
-          userInputRequirements.forEach(userInputRequirement => {
-            userInputs[userInputRequirement] = 1
-          })
-          item.userInput.forEach(property => {
-            property = property.split(':')
-            if (userInputs[property[0].trim()] && property[1]) {
-              userInputs[property[0]] = property[1].trim()
-            }
-          })
-          Object.keys(userInputs).forEach(userInput => {
-            if (!userInputs[userInput] || userInputs[userInput] === 1) {
-              delete userInputs[userInput]
-              propertiesString += ` ${userInput}`
+    const { action, properties } = productProperties
+    // * const { action, memberInputRequirements } = productProperties
+
+    // Properties are the placeholders for the values that are passed to the function. In the future will have a better name.
+
+    const memberInputRequirements = properties
+    let memberInput
+    if (member.inputs) memberInput = member.inputs.split(',')
+
+    if (Array.isArray(memberInputRequirements) && memberInputRequirements.length >= 0 && Array.isArray(memberInput)) {
+      module.exports.processMemberInputs(memberInput, memberInputRequirements, (processedInputs) => {
+        if (Object.prototype.hasOwnProperty.call(processedInputs, 'code')) { if (callback) callback(processedInputs) }
+        processAndExecuteAction(processedInputs)
+      })
+    } else {
+      processAndExecuteAction()
+    }
+
+    function processAndExecuteAction (processedInputs) {
+      if (action && Object.prototype.hasOwnProperty.call(action, 'type')) {
+        switch (action.type) {
+          case 'sendMessage': {
+            if (Object.prototype.hasOwnProperty.call(action, 'message') && Object.prototype.hasOwnProperty.call(action, 'channel')) {
+              guild.channels.fetch(action.channel).then(channel => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/alexandrumuntenas/pingu/pullRequest?prid=8657799)
```suggestion
                          guild.channels.fetch(action.channel).then((channel) => {
```